### DISCUSSION TITLE
Public facing vs Admin facing apis

### DISCUSSION THREAD

**Discussion by cipriancaba** [maoosi/prisma-appsync#141] [Q&A]
Been trying to figure out an elegant solution for creating 2 different graphql endpoints. One would be the one with CRUD operations, which will be admin facing and the other would be public(client) facing and will just contain a subset of the operations and potentially some custom resolvers.

The complication comes from the fact that I need to use one prisma schema and the generated graphql schemas would obviously be different. Also would need to disable the crud resolvers altogether in one of the instances.

Any idea on how I could achieve this? 

Been using a gateway before that would proxy the graphql requests and only expose a subset of the schema, but would like to avoid this if possible.

**Comment by maoosi** [comment]
@cipriancaba If you want to keep your endpoints and GraphQL schemas separate, then you'd probably have to create 2 distinct Prisma-AppSync projects under "public" and "private":

```
/src
  /public
  /private
  /shared
```

You could then add a tool like [prisma-merge](https://github.com/inside-labs/prisma-merge) to your workflow to merge “shared + private” and “shared + public” schemas. Similarly, you could have all your Prisma-AppSync shared business logic (hooks, resolvers, ...) inside the “shared” folder and import things as needed.

**Reply by cipriancaba** [reply]
@maoosi  Thanks for the suggestions. I ended up rolling a prisma-appsync fork and now I have 2 generators that output in different folders. Had to implement a skipModels param to skip the generation of the models
```
skipModels       = true
```
I have now 2 generated folders inside prisma:
```
prisma/generated/prisma-appsync
prisma/generated/prisma-appsync-public
```

Would there be any value you reckon adding the skipModels into `prisma-appsync`? Happy to open a pr for you to review.

**Reply by cipriancaba** [reply]
Could be potentially tweaked to include a regexp maybe so you could specify skipping just some of the models

**Reply by maoosi** [reply]
This is quite a particular use case, so I guess it depends on how much more it adds to the library from an ongoing maintenance perspective. Curious to see an example first, if you have something you can share?

**Reply by cipriancaba** [reply]
Opened #142 for example
